### DISCUSSION TITLE
MLX on linux?

### DISCUSSION THREAD

**Discussion by douglasdew** [ml-explore/mlx#544] [General]
While reading a comment made by @awni in PR #537 about isort breaking on linux, I was surprised to read that linux is a concern.

I had thought that the UMA assumption of MLX would preclude the use of MLX on the non-UMA hardware which is typical for linux.

What is the envisioned usage of MLX on linux?

**Comment by awni** [comment]
Very little linux usage. It's mostly so we can run tests on CI systems that don't have great support for MacOS.

**Reply by awni** [reply]
But even that is pretty limited since it's all CPU only...

**Reply by kungfooman** [reply]
Can I use MLX to fine-tune on Linux?

**Reply by awni** [reply]
In theory yes but for larger models it will be very slow without a GPU.
